src/o365/spfx/commands/project/project-externalize.ts

```diff
@@ -0,0 +1,262 @@
+import commands from '../../commands';
+import Command, {
+  CommandOption, CommandError, CommandAction, CommandValidate
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as os from 'os';
+import { Utils } from './project-upgrade/';
+import { Project, ExternalConfiguration } from './project-upgrade/model';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+import rules = require('./project-externalize/DefaultRules');
+import { BasicDependencyRule } from './project-externalize/rules';
+import { ExternalizeEntry } from './project-externalize/model/ExternalizeEntry';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  outputFile?: string;
+}
+
+class SpfxProjectExternalizeCommand extends Command {
+  private projectVersion: string | undefined;
+  private projectRootPath: string | null = null;
+  private supportedVersions: string[] = [
```

Comment:
Is there a reason we're not supporting earlier project versions or you just haven't tested it?

Reply:
I haven't tested it. Also, I couldn't recall whether externals were supported from from v1 and structure remained unchanged since (always in config/config.json:Externals with the same structure) or whether it was introduced later/has changed.

Reply:
We could check the structure of the previous versions. It would be good to have it, to support on-prem projects as wel.

Reply:
So basically go with all versions and fail if the structure is not what we expect? Ok I'll work on that